packages/sanity/src/tasks/src/tasks/components/sidebar/TasksListTabs.tsx

```diff
@@ -0,0 +1,97 @@
+import {TabList} from '@sanity/ui'
+import {type Dispatch, type SetStateAction, useCallback, useEffect, useMemo, useState} from 'react'
+import {type CSSProperties} from 'styled-components'
+
+import {Tab} from '../../../../../ui-components'
+import {useTasks} from '../../context'
+import {type SidebarTabsIds} from './types'
+
+const LIST_STYLES: CSSProperties = {marginLeft: '-0.5em'}
+
+interface TasksListTabsProps {
+  activeTabId: string
+  onChange: Dispatch<SetStateAction<SidebarTabsIds>>
+}
+
+interface TasksListTab {
+  id: SidebarTabsIds
+  label: string
+  isDisabled?: boolean
+}
+
+/**
+ * @internal
+ */
+export function TasksListTabs({activeTabId, onChange}: TasksListTabsProps) {
+  const {activeDocumentId} = useTasks()
+  const [isDisabledTab, setIsDisabledTab] = useState<boolean>(!activeDocumentId)
+
+  useEffect(() => {
```

Comment:
@sjelfull do you think there's other way we could handle this? Making it in a useEffect with a timeout feels a bit brittle to me.